olio/service/base.go

```diff
@@ -73,3 +73,8 @@ func (service *OlioBaseService) Stop() {
 	}
 	service.server.Stop()
 }
+
+func (service *OlioBaseService) AddVersionProvider(versionExtractor ...VersionExtractor) {
+	versionResource := olioResources.NewVersionResource(versionExtractor)
```

Comment:
we should have the version resource available even though they didn't inject a version provider. the creation of this resource needs to happen elsewhere; all this should do is inject the versionExtractor into it